compliance/collection/nodescanv2/nodescan.go

```diff
@@ -16,5 +19,60 @@ type NodeScan struct {
 
 // Scan scans the current node and returns the results as storage.NodeInventory object
 func (n *NodeScan) Scan(nodeName string) (*storage.NodeInventory, error) {
-	return nil, errors.New("Not implemented")
+	componentsHost, err := nodes.Analyze(nodeName, "/host/", false)
+	log.Info("Finished node inventory /host scan")
+	if err != nil {
+		log.Errorf("Error scanning node /host inventory: %v", err)
+	}
+	log.Infof("Components found under /host: %v", componentsHost)
+	if err != nil {
+		return nil, err
+	}
+
+	var protoComponents *scannerV1.Components
+	if componentsHost != nil {
+		protoComponents = protoComponentsFromScanComponents(componentsHost)
+	}
+	m := &storage.NodeInventory{
+		NodeName:   nodeName,
+		ScanTime:   timestamp.TimestampNow(),
+		Components: protoComponents,
+	}
+	return m, nil
+}
+
+func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Components {
+	var components []*scannerV1.RHELComponent
+	// For now, we only care about RHEL components, but this must be extended once we support non-RHCOS
+	if c.CertifiedRHELComponents != nil {
+		components = convertRHELComponents(c.CertifiedRHELComponents)
+	}
```

Comment:
Two suggestions:
1. Handle the case of `c == nil`.
2. Move `if c.CertifiedRHELComponents != nil` into `convertRHELComponents`.

```suggestion
	if c == nil {
		return nil
	}
	// For now, we only care about RHEL components, but this must be extended once we support non-RHCOS
	rhelComponents := convertRHELComponents(c.CertifiedRHELComponents)
```